plugins/action/system/shutdown.py

```diff
@@ -0,0 +1,234 @@
+            module_name='ansible.legacy.find',
+            module_args={
+                'paths': search_paths,
+                'patterns': [shutdown_bin],
+                'file_type': 'any'
+            }
+        )
+
+        full_path = [x['path'] for x in find_result['files']]
+        if not full_path:
+            raise AnsibleError('Unable to find command "{0}" in search paths: {1}'.format(shutdown_bin, search_paths))
+        self._shutdown_command = full_path[0]
+        return self._shutdown_command
+
+    def deprecated_args(self):
+        for arg, version in self.DEPRECATED_ARGS.items():
+            if self._task.args.get(arg) is not None:
+                display.warning("Since Ansible {version}, {arg} is no longer a valid option for {action}".format(
+                    version=version,
+                    arg=arg,
+                    action=self._task.action))
+
+    def perform_shutdown(self, task_vars, distribution):
+        result = {}
+        shutdown_result = {}
+        shutdown_command = self.get_shutdown_command(task_vars, distribution)
+        shutdown_command_args = self.get_shutdown_command_args(distribution)
+        shutdown_command_exec = '{0} {1}'.format(shutdown_command, shutdown_command_args)
+
+        self.cleanup(force=True)
+        try:
+            display.vvv("{action}: shutting down server...".format(action=self._task.action))
+            display.debug("{action}: shutting down server with command '{command}'".format(action=self._task.action, command=shutdown_command_exec))
+            shutdown_result = self._low_level_execute_command(shutdown_command_exec, sudoable=self.DEFAULT_SUDOABLE)
+        except AnsibleConnectionFailure as e:
+            # If the connection is closed too quickly due to the system being shutdown, carry on
+            display.debug('{action}: AnsibleConnectionFailure caught and handled: {error}'.format(action=self._task.action, error=to_text(e)))
+            shutdown_result['rc'] = 0
+
+        if shutdown_result['rc'] != 0:
+            result['failed'] = True
+            result['shutdown'] = False
+            result['msg'] = "Shutdown command failed. Error was {stdout}, {stderr}".format(
+                stdout=to_native(shutdown_result['stdout'].strip()),
+                stderr=to_native(shutdown_result['stderr'].strip()))
+            return result
+
+        result['failed'] = False
+        return result
+
+    def run(self, tmp=None, task_vars=None):
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        # If running with local connection, fail so we don't shutdown ourself
+        if self._connection.transport == 'local':
+            msg = 'Running {0} with local connection would shutdown the control node.'.format(self._task.action)
+            return {'changed': False, 'elapsed': 0, 'shutdown': False, 'failed': True, 'msg': msg}
+
+        if self._play_context.check_mode:
```

Comment:
How about running everything when `self._play_context.check_mode` is `True` except the `shutdown_result = self._low_level_execute_command(shutdown_command_exec, sudoable=self.DEFAULT_SUDOABLE)` in the previous function? That way, check mode is even more realistic.

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/700/commits/e78ae50651498ad4efb7525dfcd4c49cf40bb467.